Fix - update .NET 4.7.1 configBuilders integration examples

seangwright opened the pull request:
Change secretsFile to userSecretsFile
Update library names/namespaces/nuget package versions
Format secrets xml example

This doc is pretty out of date and I kept forgetting **not** to reference it when testing out configuration builder stuff with .net 4.7.1. The examples are incorrect with the latest `preview` nuget packages so I thought I would update them so that someone can follow along and end up with a functioning example.

azurecla commented:
Hi __@sahaj08__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (sahajs). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

rmca14 commented:
@barbkess 
Please review and approve/disapprove here, thanks.

rmca14 commented:
@JennieHubbard 
Please review and approve/disapprove with a comment here, thanks.

JennieHubbard commented:
#sign-off

seangwright committed:
Fix - update .NET 4.7.1 configBuilders integration

Change secretsFile to userSecretsFile
Update library names/namespaces/nuget package versions
Format secrets xml example

seangwright committed:
Feat - configBuilders examples local secrets

Add format example for an app local secrets file

StephenMolloy commented on articles/key-vault/vs-secure-secret-appsettings.md:
````diff
@@ -97,18 +97,18 @@ If you are writing a quick prototype and don't want to provision Azure resources
 1. Install the following NuGet package to your project
     ```
-    Microsoft.Configuration.ConfigurationBuilders.Basic.1.0.0-alpha1.nupkg
+    Microsoft.Configuration.ConfigurationBuilders.UserSecrets.1.0.0-preview.nupkg
````
-preview2 is the latest... although we hope to have preview3 out shortly, which will have some added features for UserSecrets in a VS scenario.

StephenMolloy commented on articles/key-vault/vs-secure-secret-appsettings.md:
````diff
@@ -143,7 +145,7 @@ Follow instructions from ASP.NET core section to configure a Key Vault for your
 1. Install the following NuGet package to your project
 ```
-Microsoft.Configuration.ConfigurationBuilders.Azure.1.0.0-alpha1.nupkg
+Microsoft.Configuration.ConfigurationBuilders.Azure.1.0.0-preview.nupkg
````
Again, -preview2 is the latest at the moment... and it looks like the latest version of this doc has already been updated for Azure Key Vault.

StephenMolloy commented on articles/key-vault/vs-secure-secret-appsettings.md:
```diff
@@ -154,7 +156,7 @@ Microsoft.Configuration.ConfigurationBuilders.Azure.1.0.0-alpha1.nupkg
     <configBuilders>
         <builders>
-            <add name="KeyVault" vaultName="Test911" type="Microsoft.Configuration.ConfigurationBuilders.AzureKeyVaultConfigBuilder, ConfigurationBuilders, Version=1.0.0.0, Culture=neutral" />
+            <add name="KeyVault" vaultName="Test911" type="Microsoft.Configuration.ConfigurationBuilders.AzureKeyVaultConfigBuilder, Microsoft.Configuration.ConfigurationBuilders.Azure, Version=1.0.0.0, Culture=neutral" />
```
It might be worth mentioning the 'connectionString' property as a fallback here if VS isn't set up with the proper MSI/credentials for the connection to happen automagically.
